tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -549,9 +553,29 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(server_conn->actual_protocol_version, S2N_TLS13);
         EXPECT_EQUAL(client_conn->actual_protocol_version, S2N_TLS13);
 
-        EXPECT_SUCCESS(s2n_connection_free(server_conn));
-        EXPECT_SUCCESS(s2n_connection_free(client_conn));
-        EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
+        /* since TLS 1.3 was negotiated, and the client hasn't called recv yet
+         * there should be no session ticket available. 512 is an arbitrary
+         * number, nothing should be copied because this operation should fail
+         */
+        EXPECT_EQUAL(s2n_connection_get_session_length(client_conn), 0);
+        /* In the event that get_session actually writes data, we want an error,
+         * not a segfault.
+         */
+        s2n_realloc(&ticket, 512);
+        EXPECT_EQUAL(s2n_connection_get_session(client_conn, ticket.data, 512), 0);
+
+        /* the server automatically sends a session ticket as part of completing
+         * handshake io, so it isn't necessary to call send again
+         */
+        uint8_t data = 1;
+        s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+        EXPECT_FAILURE_WITH_ERRNO(s2n_recv(client_conn, &data, 1, &blocked), S2N_ERR_IO_BLOCKED);
+
+        /* Get the TLS1.3 session ticket */
+        size_t tls13_session_ticket_len = s2n_connection_get_session_length(client_conn);
+        EXPECT_SUCCESS(s2n_blob_zero(&ticket));
+        EXPECT_SUCCESS(s2n_realloc(&ticket, tls13_session_ticket_len));
+        EXPECT_SUCCESS(s2n_connection_get_session(client_conn, ticket.data, ticket.size));
```

Comment:
I'm still not sure you're gaining anything from this.

Reply:
The problem with the client session ticket not being cleared is that it poisoned the state for new ticket issuance, therefore doing a resumption handshake is a check that we have fixed that problem.

I understand that this particular case wasn't failing before (e.g. we behaved correctly in this situation as long as you didn't call get_session before having received a ticket) but this felt like a low-cost way to get some extra coverage. But it seems like we're comfortable with our test coverage in this area, so I'll go ahead and remove it.